example/test/function_component_test.dart

```diff
@@ -183,6 +183,53 @@ UseRefTestComponent(Map props) {
   ]);
 }
 
+int fibonacci(int n) {
+  if (n <= 1) {
+    return 1;
+  }
+  return fibonacci(n - 1) + fibonacci(n - 2);
+}
+
+final useMemoTestFunctionComponent = react.registerFunctionComponent(UseMemoTestComponent, displayName: 'useMemoTest');
+
+UseMemoTestComponent(Map props) {
+  final reRender = useState(0);
+  final num = useState(35);
+
+  final fib = useMemo(
+    () {
+      print('calculating fibonacci...');
+      return fibonacci(num.value);
+    },
+
+    /// This dependency prevents [fib] from being re-calculated every time the component re-renders.
+    [num.value],
+  );
+
+  return react.Fragment({}, [
+    react.div({'key': 'div'}, ['Fibonacci of ${num.value} is $fib']),
+    react.button({'key': 'button1', 'onClick': (_) => num.setWithUpdater((prev) => prev + 1)}, ['+']),
+    react.button({'key': 'button2', 'onClick': (_) => reRender.setWithUpdater((prev) => prev + 1)}, ['re-render']),
+  ]);
+}
+
+final useMemoTestFunctionComponent2 =
+    react.registerFunctionComponent(UseMemoTestComponent2, displayName: 'useMemoTest2');
+
+UseMemoTestComponent2(Map props) {
+  final reRender = useState(0);
+  final num = useState(35);
```

Comment:
#nit probably shouldn't use `num` as a variable name - since its a reserved word / primitive type in Dart's type system.